charts/posthog/templates/migrate.job.yaml

```diff
@@ -33,12 +33,16 @@ spec:
       containers:
       - name: migrate-job
         image: {{ template "posthog.image.fullPath" . }}
+        # switch over to using /bin/migrate once 1.34 is released - https://github.com/PostHog/posthog/pull/8872
         command:
           - /bin/sh
           - -c
           - |
+            set -e
             python manage.py notify_helm_install || true
-            python manage.py migrate && python manage.py migrate_clickhouse
+            python manage.py migrate
+            python manage.py migrate_clickhouse
+            python manage.py run_async_migrations --check
```

Comment:
What's the deal with only having `--check` here where as the others don't? Worth adding a comment with the reasoning as at first glance if looks like a mistake.

Reply:
Is there another place that the migrations are actually run? If not then I think we'll be in a deadlock if we have migrations to apply, i.e. init container will never complete successfully

Reply:
Async migrations can take a long time (e.g. 5h) which is why we don't want to run them during the helm upgrade. They need to be ran beforehand. If they haven't been completed we'd see an error (it will also be a major upgrade note if they need to be completed). We do need to run the django & CH migrations before to (1) be able to run the check (2) some of the async migrations would be auto-completed (note we do auto-complete them if we can within `--check`). If you have better ideas how to do this I'm all ears.

Reply:
Yes, the task would wait forever & end up doing a rollback if `--atomic` was used (see description)